[RNMobile] [Embed block] Fix content disappearing on Android when switching light/dark mode

fluiddot opened the pull request:
`gutenberg-mobile` PR: https://github.com/wordpress-mobile/gutenberg-mobile/pull/3859

<!-- Learn the overall process and best practices for pull requests at https://github.com/WordPress/gutenberg/blob/HEAD/docs/contributors/repository-management.md#pull-requests. -->

<!-- Gutenberg's license is in the process of updating to be dual-licensed under the GPL and MPL. As part of that transition, all new contributions are dual-licensed. For more information, see: https://github.com/WordPress/gutenberg/blob/trunk/LICENSE.md -->

## Description
<!-- Please describe what you have changed or added -->

As it was done for the device rotation action, I'm adding to prop `key` of the SandBox component the value of the light/dark mode, this way when this device setting changes it will recreate the WebView to prevent the content from disappearing.

## How has this been tested?
<!-- Please describe in detail how you tested your changes. -->
<!-- Include details of your testing environment, tests ran to see how -->
<!-- your change affects other areas of the code, etc. -->

1. Open the app on Android (connect the app to a local Metro server as the previews only work in development mode)
2. Open a post/page
3. Add an embed block
4. Add a valid embed URL
5. Observe that the embed preview is displayed
6. Switch to light or dark mode
7. Observe that the embed preview is still displayed

## Screenshots <!-- if applicable -->

N/A

## Types of changes
<!-- What types of changes does your code introduce?  -->
<!-- Bug fix (non-breaking change which fixes an issue) -->
<!-- New feature (non-breaking change which adds functionality) -->
<!-- Breaking change (fix or feature that would cause existing functionality to not work as expected) -->
Bug fix

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. <!-- Check code: `npm run lint`, Guidelines: https://developer.wordpress.org/coding-standards/wordpress-coding-standards/javascript/ -->
- [x] My code follows the accessibility standards. <!-- Guidelines: https://developer.wordpress.org/coding-standards/wordpress-coding-standards/accessibility/ -->
- [ ] I've tested my changes with keyboard and screen readers. <!-- Instructions: https://github.com/WordPress/gutenberg/blob/HEAD/docs/contributors/accessibility-testing.md -->
- [x] My code has proper inline documentation. <!-- Guidelines: https://developer.wordpress.org/coding-standards/inline-documentation-standards/javascript/ -->
- [x] I've included developer documentation if appropriate. <!-- Handbook: https://developer.wordpress.org/block-editor/ -->
- [x] I've updated all React Native files affected by any refactorings/renamings in this PR (please manually search all `*.native.js` files for terms that need renaming or removal). <!-- React Native mobile Gutenberg guidelines: https://github.com/WordPress/gutenberg/blob/HEAD/docs/contributors/code/native-mobile.md -->

fluiddot committed:
Add dark/light mode value in sandbox key for Android

ceyhun commented on packages/block-library/src/embed/embed-preview.native.js:
```diff
@@ -137,4 +140,4 @@ const EmbedPreview = ( {
 };
 
-export default memo( EmbedPreview );
+export default memo( withPreferredColorScheme( EmbedPreview ) );
```
Seems like we're only using `withPreferredColorScheme` to get and pass down `preferredColorScheme` to `SandBox`. Maybe we can instead use `usePreferredColorSchemeStyle` hook in `SandBox` component directly?

fluiddot replied:
Yeah although the `usePreferredColorSchemeStyle` hook is meant to be used for picking the right style depending on the light/dark mode status. In this case, I'd rather use `usePreferredColorScheme` that returns the color scheme instead 😃 .

fluiddot replied:
This has been addressed in [this commit](https://github.com/WordPress/gutenberg/pull/34207/commits/279c13714ec934a85383ffddc0013d949efca4f7).

ceyhun replied:
Do you think it would be possible/better to use the hook directly in `SandBox` and not make any changes in `EmbedPreview` component?

fluiddot replied:
Yeah, I initially thought to add it there but since `SandBox` is a generic component I wanted to leave the color scheme calculation open to the parent. However, I agree that looks cleaner to have the hook directly there.

fluiddot replied:
@ceyhun I applied your suggestion in [this commit](https://github.com/WordPress/gutenberg/pull/34207/commits/39274fba75b830e52c7cbb3c121db09f12b8838f). I also updated the branch with the last changes from `trunk` and solve conflicts, the PR is ready for another review 🙇 .

fluiddot committed:
Use usePreferredColorScheme hook

fluiddot commented:
@ceyhun I've applied the suggestion from [your comment](https://github.com/WordPress/gutenberg/pull/34207#discussion_r701815993), so the PR is ready for another review 🎊 .

fluiddot committed:
Merge branch 'trunk' into rnmobile/embed-block-fix-android-dark-mode

# Conflicts:
#	packages/block-library/src/embed/embed-preview.native.js
#	packages/components/src/sandbox/index.native.js

fluiddot committed:
Fetch color scheme in sandbox component directly

fluiddot committed:
Update react-native-editor changelog

Additionally I added a a missing entry related to the WP embed component implementation.

fluiddot commented on packages/react-native-editor/CHANGELOG.md:
```diff
@@ -10,6 +10,8 @@ For each user feature we should also add a importance categorization label  to i
 -->
 
 ## Unreleased
+-   [**] [Embed block] Implement WP embed preview component [#34004]
```
I added the entry for the WP embed preview component implementation because it was missing.

ceyhun reviewed (approved):
:shipit:
